specification/search/data-plane/Azure.Search/preview/2020-06-30-Preview/searchservice.json

```diff
@@ -5789,13 +5789,246 @@
         },
         "configuration": {
           "type": "object",
+          "properties": {
```

Comment:
I don't know if this actually works, though. OpenAPI spec says nesting property definitions is not supported, so I don't know if our language generators will work with this. It might be worth trying one or all of them against this PR before we merge, lest we break them at worst.

Reply:
Per internal discussion, could we actually make this a separate model? That would provide some flexibility in mapping a new type with explicit properties into a dictionary for the SDKs, while benefitting from the generators not trying to change the type (which is a breaking change anyway). The model type would still have the same properties here - including support for additional properties - so it wouldn't require any service implementation changes.

Reply:
Done in commit https://github.com/Azure/azure-rest-api-specs/pull/10349/commits/e9ae5e749b606bf411dccfba3740fac7357f00dd